auth/api/grpc/requests.go

```diff
@@ -34,7 +34,7 @@ type issueReq struct {
 
 func (req issueReq) validate() error {
 	if req.email == "" {
-		return errors.ErrUnauthorizedAccess
+		return errors.ErrAuthentication
```

Comment:
@manuio if I understood @dusanb94 well and per what we wrote here: https://github.com/mainflux/mainflux/pull/1538, `ErrUnauthorizedAccess` is an authorization, not authentication error, and should be replaced by `ErrAuthorization` and return `403 Forbidden`.

@dusanb94 did I get this right?

Reply:
@drasko Check your previous remark. `401 Unauthorized` is currently  `errors.ErrUnauthorizedAccess`. I'm relacing it by `errors.ErrAuthentication`

Reply:
@manuio comment was:
> Change `ErrAuthorization` to a `ErrAuthentication` and `ErrUnauthorizedAccess` to `ErrAuthorization`

What we want to accomplish is: when there is an authentication error, use our `ErrAuthentication`, and associate `401 Unauthorized` with it. When error is due to the authorization (i.e. client authenticated, but not authorizaed), use `ErrAuthorization` and `403 Forbidden`.

Reply:
I think that there is a confusion here. You proposed this:
```

    ErrAuthentication -> 401 Unauthorized
    ErrAuthorization -> 403 Forbidden
```

What you are proposing here is worng:
```
Change ErrAuthorization to a ErrAuthentication and ErrUnauthorizedAccess to ErrAuthorization
```

What I did is: `ErrAuthentication` as `401 - StatusUnauthorized` and `ErrAuthorization` as `403 - StatusForbidden`. 
So `ErrUnauthorizedAccess` have to be renamed as `ErrAuthentication` not `ErrAuthorization`, right?

Reply:
It should be like this: 

- Missing or invalid JWT -> errors.ErrAuthentication -> Status Code **Unauthorized**
- Policies evaluation unsuccessful -> errors.ErrAuthorization -> Status Code **Forbidden**

Reply:
@drasko `Change ErrAuthorization to a ErrAuthentication and ErrUnauthorizedAccess to ErrAuthorization.` was confusing for @manuio because of the error messages. Let's stick to [this](https://github.com/mainflux/mainflux/pull/1538#discussion_r794324890) and we should be OK.